test_allianceutils/tests/profile_auth/models.py

```diff
@@ -1,16 +1,38 @@
-from authtools.models import AbstractEmailUser
-from authtools.models import UserManager
+import django.contrib.auth.models as auth_models
 from django.db import models
 
 import allianceutils.auth.models
 
 
-class UserManager(allianceutils.auth.models.GenericUserProfileManagerMixin, UserManager):
+# We deliberately use a model with a non-standard USERNAME_FIELD to make sure
+# that nothing is hardcoded to expect 'username' as the username field
+class UserManager(allianceutils.auth.models.GenericUserProfileManagerMixin, auth_models.UserManager):
     def get_by_natural_key(self, username):
         return self.get(email=username)
 
+    def get_by_natural_key(self, username: str) -> models.Model:
+        username = self.normalize_email(username)
+        return super().get_by_natural_key(username)
+        return self.get(**{self.model.USERNAME_FIELD: username})
+
+    def create_superuser(self, **kwargs):
+        user = self.create_user(is_staff=True, is_superuser=True, **kwargs)
+        user.save(using=self._db)
+        return user
+
+    def create_user(self, email, password=None, is_staff=False, **kwargs):
+        email = self.normalize_email(email)
+        user = self.model(email=email,
+            # is_staff=is_staff,
```

Comment:
should make this behavior more obvious to people??

Reply:
ie docstr/readme mention it

Reply:
This is only for the internal test model, it won't affect anything touching this project.

I've fixed it anyway though in case someone copies the code.

(Originally `is_staff` wasn't present at all but some of the ancestor classes were using it and it was easier to include it rather than try to fight them.)